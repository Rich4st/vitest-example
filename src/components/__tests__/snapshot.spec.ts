import { describe, it, expect, beforeAll, afterAll } from 'vitest';


describe.skip('snapshot test', () => {
  const target = [
    {
      name: 'Orange',
      price: 100,
    },
    {
      name: 'Apple',
      price: 200,
    },
  ]

  beforeAll(() => {
    expect(target).toContainEqual({ name: 'Orange', price: 100 })
    console.log('ðŸš€ -----> test start')
  })

  afterAll(() => {
    console.log('ðŸš€ -----> test end')
  })

  // snapshot file
  it.skip('fruit snapshot', () => {

    expect(target).toMatchSnapshot();
  })
  // math snapshot inline
  it('math napshot inline', () => {

    expect(target).toMatchInlineSnapshot([
      {
        name: 'Orange',
        price: 100,
      },
      {
        name: 'Apple',
        price: 200,
      },
    ])
  })

  // throw error test
  it('throw error', () => {
    const food = (food: string): string => {
      throw new Error(`I don't like ${food}`);
    }

    //expect(food('apple')).toThrowErrorMatchingSnapshot(); // error
    expect(() => food('è‹¦ç“œ')).toThrowError('I don\'t like è‹¦ç“œ'); // pass
  })
})
