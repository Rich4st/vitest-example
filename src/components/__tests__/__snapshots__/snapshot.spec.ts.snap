// Vitest Snapshot v1

exports[`snapshot test > fruit snapshot 1`] = `
[
  {
    "name": "Orange",
    "price": 100,
  },
  {
    "name": "Apple",
    "price": 2000,
  },
]
`;
